### PR TITLE
fix minor issue with resolv_conf template

### PR DESCRIPTION
**Pull request by andreaf74**
This fixes a little problem with the template used by resolv_conf module, when more than one non boolean options is used, resulting in wrong formatting, like this:

options rotate single-request attempts:1
timeout:1

I've modified the template in order to put all the options (boolean and not boolean) on the same line.

### CODE REVIEW

File: templates/resolv.conf.tmpl [hunk 1]
```diff
@@ -23,8 +23,5 @@ sortlist {% for sort in sortlist %}{{sort}} {% endfor %}
 {% endif %}
 {% if options or flags %}
 
-options {% for flag in flags %}{{flag}} {% endfor %}
-{% for key, value in options.items() -%}
- {{key}}:{{value}}
-{% endfor %}
+options {% for flag in flags %}{{flag}} {% endfor %}{% for key, value in options.items() %}{{key}}:{{value}} {% endfor %}
```
Comment (igalic) [inline templates/resolv.conf.tmpl]: perhaps this would be easier to read

File: templates/resolv.conf.tmpl [hunk 2]
```diff
@@ -23,8 +23,5 @@ sortlist {% for sort in sortlist %}{{sort}} {% endfor %}
 {% endif %}
 {% if options or flags %}
 
-options {% for flag in flags %}{{flag}} {% endfor %}
```
Comment (igalic) [inline templates/resolv.conf.tmpl]: if we ended this line in

```python
{% endfor -%}
```

File: templates/resolv.conf.tmpl [hunk 3]
```diff
@@ -23,8 +23,5 @@ sortlist {% for sort in sortlist %}{{sort}} {% endfor %}
...
-{% for key, value in options.items() -%}
```
Comment (igalic) [inline templates/resolv.conf.tmpl]: or if we started this line with 

```python
{%- for %}
```

Comment (andreaf74) [reply templates/resolv.conf.tmpl]: Pushed a new commit to improve readability.
Does it feel better?
Thanks.

**Comment by igalic** [comment]
Nice! 

how about you add a test, so we know what this is fixing, and that it works

**Comment by blackboxsw** [comment]
Thank you so much for your contribution to cloud-init and for making cloud-init better!

Contributions to cloud-init are governed by a Contributor License Agreement (CLA). Please confirm that you have signed the CLA per our [hacking guide](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html#do-these-things-once) and run migrate-lp-user-to-github to associate this github account as someone who has signed the CLA.

File: templates/resolv.conf.tmpl [hunk 4]
```diff
@@ -23,8 +23,12 @@ sortlist {% for sort in sortlist %}{{sort}} {% endfor %}
...
- {{key}}:{{value}}
+options
```
Comment (OddBloke) [inline templates/resolv.conf.tmpl]: It was not immediately obvious to me (as a developer/user on a distro that doesn't use resolvconf) that the intended behaviour here is for all the flags and options to be on the same line. Could we possibly add a `{# comment #}` to note that, so future readers have an easier time grokking what's happening here?

Comment (andreaf74) [reply templates/resolv.conf.tmpl]: Hi,
Just pushed the requested change.
Please let me know if it works for you.

Thanks!

**Pull request merged**